pkg/config/config.go

```diff
@@ -1260,11 +1260,6 @@ func setNumWorkers(config Config) {
 		numWorkers = 1
 	}
 
-	if numWorkers > MaxNumWorkers {
```

Comment:
Do we still need the `MaxNumWorkers` variable ?

Reply:
We might still want to print a warning or something telling users that this will have impact, so it's not enable in the wrong case

Reply:
> Do we still need the `MaxNumWorkers` variable ?

Yes, [here](https://github.com/DataDog/datadog-agent/blob/e081bed84866686eeab56a98da5ff9b4b9f03ded/pkg/collector/runner/runner.go#L111-L124). I think it's good to keep the limit 25 for this logic setting # of workers based on the # of check instances.

> We might still want to print a warning or something telling users that this will have impact, so it's not enable in the wrong case

Agree with having a clear message. It seems warning level might a bit too high since it will be a normal use case to set a # of check runner >25 if this PR is merged.

I can add a `info` for >25 runners to warn people, e.g. >25 might not work well if they are running many Python checks

We currently already log here the # of check runners here: https://github.com/DataDog/datadog-agent/blob/e081bed84866686eeab56a98da5ff9b4b9f03ded/pkg/collector/runner/runner.go#L92

Reply:
Thanks for the answer. I think it's fine then. We might want to add a warning later if we have too many support cases about python performances with a high number of workers.